scripts/scr_poststage.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+#
+# SCR allows you to spawn off dataset transfers "in the background"
+# that will finish some time after a job completes.  This saves you from
+# using your compute time to transfer datasets.  You can do this by
+# specifying XFER=BBAPI_POSTSTAGE for a storage descriptor in scr.conf.
+# Currently, this is only supported on on IBM burst buffer nodes.
+#
+# This script is to be run as a 2nd-half post-stage script on an IBM
+# system.  A 2nd-half post-stage script will run after all the job's burst
+# buffers transfers have finished (which could be hours later after the job
+# finishes).  This script will finalize any completed burst buffer dataset
+# transfers so that they're visible to SCR.
+#
+
+# Please fill in BINDIR AND PREFIX below:
+
+# The full path to the dir containing the SCR binaries (scr_index and scr_flush_file)
+BINDIR=""
+
+# The dir path to your SCR PREFIX
+PREFIX="/p/gpfs1/`whoami`"
+
+# Path to where you want the scr_poststage.sh log
+LOGFILE=$PREFIX/scr_poststage.log
+
+if [ ! -d "$BINDIR" ] ; then
+	echo "Please specify BINDIR in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+if [ ! -d "$PREFIX" ] ; then
+	echo "Please specify PREFIX in scr_poststage.sh" >> $LOGFILE
+	exit
+fi
+
+export PATH="$PATH:$BINDIR"
+
+function do_poststage {
+	echo "$(date '+%m/%d/%y %H:%m:%S') Begin post_stage" >> $LOGFILE
+	echo "Current index before finalizing:" >> $LOGFILE
+	scr_index -l -p $PREFIX >> $LOGFILE
+
+	# Finalize each dataset listed in the flush file
+	for ID in $(scr_flush_file --dir $PREFIX --list-ckpt) ; do
+		DSET="$(scr_flush_file --dir $PREFIX --name $ID)"
+
+		echo "Looking at dataset $ID ($DSET)" >> $LOGFILE
+
+		if ! scr_flush_file --dir $PREFIX --need-flush $ID &>> $LOGFILE ; then
+			# Dataset is already flushed, skip it
+			continue
+		fi
+
+		echo "Finalizing transfer for dataset $ID ($DSET)" >> $LOGFILE
+		if ! scr_flush_file -r -s $ID --dir $PREFIX &>> $LOGFILE ; then
+			echo "Error: Can't resume dataset $ID, rc=$?" >> $LOGFILE
+			continue
+		fi
```

Comment:
Thanks for taking a look at how pthread transfers are handled in ``SCR_Finalize``.  The other spot we want to account for a pthread transfer is here.  Perhaps the job started a pthread flush but died or ran out of time before it could be finalized within the library.  In that case, I think we'll fall through to this point.  It makes sense for the resume to fail and mark that as an error as you have.  We should also verify that ``scr_flush_file`` recognizes it failed here.

Do you know whether that works?

Reply:
I don't know, let me give it a quick test and report back

Reply:
Just ran the test - the good news is that an interrupted sync transfer will not resume in `scr_poststage.sh` since the source file is gone (`AXL_Resume()` returns -1).  The bad news is that I removed the return code from resume_transfer() in review, so `scr_poststage.sh` wasn't seeing the failure.  Once I restored the return code, `scr_poststage.sh` correctly caught the error and didn't finalize the dataset.  I'll include the fix in my next push.